src/Integration/Constants.cs

```diff
@@ -15,6 +15,11 @@ public static class Constants
         public const string CodeAnalysisRuleSetPropertyKey = "CodeAnalysisRuleSet";
 
         /// <summary>
+        /// The property key which corresponds to the code analysis rule set directories
+        /// </summary>
+        public const string CodeAnalysisRuleSetDirectoriesPropertyKey = "CodeAnalysisRuleSetDirectories";
```

Comment:
Something that came to mind (forgot to mention at the top of the summary) - you can specify rule set directories that will be used when searching for your ruleset. Up until now it wasn't needed, but once we included an existing ruleset, we do need to specific this in order to be able to find the effective rules during conflicts detection.